numpy/core/src/common/npy_cpu_features.c.src

```diff
@@ -331,8 +518,7 @@ npy__cpu_init_features_linux(void)
     }
     if (hwcap == 0 && hwcap2 == 0) {
         /*
-         * FIXME: failback to compiler definitions,
-         * BTW we can parse /proc/cpuinfo for badly patched kernels
+         * failback to compiler definitions
```

Comment:
No need for duplicate parsing.  You can fill `hwcap` and `hwcap2` bits from `/proc/cpuinfo` in here, only when parsing `auxv` fail.

Reply:
reordered to getauxval  --> auxv --> cpuinfo.